op-node/rollup/derive/ecotone_upgrade_transactions.go

```diff
@@ -0,0 +1,136 @@
+package derive
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/solabi"
+)
+
+const UpgradeToFuncSignature = "upgradeTo(address)"
+
+var (
+	// known address w/ zero txns
+	l1BlockDeployerAddress        = common.HexToAddress("0x4210000000000000000000000000000000000000")
+	gasPriceOracleDeployerAddress = common.HexToAddress("0x4210000000000000000000000000000000000001")
+
+	newL1BlockAddress        = crypto.CreateAddress(l1BlockDeployerAddress, 0)
+	newGasPriceOracleAddress = crypto.CreateAddress(gasPriceOracleDeployerAddress, 0)
+
+	deployL1BlockSource        = UpgradeDepositSource{Intent: "Ecotone: L1 Block Deployment"}
+	deployGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Deployment"}
+	updateL1BlockProxySource   = UpgradeDepositSource{Intent: "Ecotone: L1 Block Proxy Update"}
+	updateGasPriceOracleSource = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Proxy Update"}
+	enableEcotoneSource        = UpgradeDepositSource{Intent: "Ecotone: Gas Price Oracle Set Ecotone"}
+	beaconRootsSource          = UpgradeDepositSource{Intent: "Ecotone: beacon block roots contract deployment"}
+
+	eip4788From         = common.HexToAddress("0x0B799C86a49DEeb90402691F1041aa3AF2d3C875")
+	eip4788CreationData = common.Hex2Bytes("0x60618060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500")
+	UpgradeToFuncBytes4 = crypto.Keccak256([]byte(UpgradeToFuncSignature))[:4]
+)
+
+func EcotoneNetworkUpgradeTransactions() ([]hexutil.Bytes, error) {
+	upgradeTxns := make([]hexutil.Bytes, 0, 5)
+
+	deployL1BlockTransaction, err := types.NewTx(&types.DepositTx{
+		SourceHash:          deployL1BlockSource.SourceHash(),
+		From:                l1BlockDeployerAddress,
+		To:                  nil,
+		Value:               nil,
+		Gas:                 300_000,
+		IsSystemTransaction: false,
+		Data:                common.FromHex(bindings.L1BlockMetaData.Bin),
```

Comment:
Is this metadata stable enough?

Reply:
this is not stable enough, we should copy in a string. we should define the string as a `const` and in the godoc show how to reproduce the string. something like: "commit ..., `jq .bytecode < forge-artifacts/...`"